txn.go

```diff
@@ -234,11 +239,12 @@ func (tr *transactionRunner) Run(transactions TransactionSource) error {
 		if err != nil {
 			return err
 		}
-		if len(ops) == 0 {
+		if len(t.Ops) == 0 {
 			// Treat this the same as ErrNoOperations but don't suppress other errors.
 			return nil
 		}
-		if err := tr.RunTransaction(ops); err == nil {
+		t.Attempt = i
```

Comment:
I feel like this might be easier to read, and harder to make a mistake if the Transaction wasn't created until here, as just:
t := Transaction{
  Ops: ops,
  Attempt: i,
}

And we leave the local vars 'ops' and 'err'.

That way there isn't any chance to accidentally still refer to the old ops in the next trip around the loop.